sdk/iot/inc/az_iot_sas_token.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef AZ_IOT_SAS_TOKEN_H
+#define AZ_IOT_SAS_TOKEN_H
+
+#include <stdlib.h>
+#include <az_result.h>
+#include <az_span.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_iot_sas_params {
+    az_span device_id;
+    az_span iothub_fqdn;
```

Comment:
You'll need some docs here; at least a comment for fqdn

Reply:
What is this structure for if the next function takes all these fields as arguments?

Reply:
This was not meant to be sent on the PR. We are avoiding using structs for params, as per insights shared by @CIPop. 
Removed.